docs/sql/sql.md

```diff
@@ -0,0 +1,440 @@
+
+Examples:
+```sql
+SELECT * FROM Inventory ORDER BY price;
+```
+
+#### LIMIT Clause
+
+```ebnf
+LIMIT INTEGER
+```
+
+`LIMIT` restricts the number of rows returned.
+
+Examples:
+```sql
+-- What are the 5 highest priced items in my inventory?
+SELECT * FROM Inventory ORDER BY price LIMIT 5;
+```
+
+### INSERT
+
+```ebnf
+INSERT INTO table [ '(' column { ',' column } ')' ] VALUES '(' literal { ',' literal } ')'
+```
+
+Examples:
+```sql
+-- Inserting one row
+INSERT INTO Inventory (item_id, item_name) VALUES (1, 'health1');
+
+-- Inserting two rows
+INSERT INTO Inventory (item_id, item_name) VALUES (1, 'health1'), (2, 'health2');
+```
+
+### DELETE
+
+```ebnf
+DELETE FROM table [ WHERE predicate ]
+```
+
+Deletes all rows from a table.
+If `WHERE` is specified, only the matching rows are deleted.
+
+Examples:
+```sql
+-- Delete all rows
+DELETE FROM Inventory;
+
+-- Delete all rows with a specific item_id
+DELETE FROM Inventory WHERE item_id = 1;
+```
+
+### UPDATE
+
+```ebnf
+UPDATE table SET [ '(' assignment { ',' assignment } ')' ] [ WHERE predicate ]
+```
+
+Updates column values of existing rows in a table.
+The columns are identified by the `assignment` defined as `column '=' expr`.
+The column values are updated for all rows that match the `WHERE` condition.
+The rows are updated after the `WHERE` condition is evaluated for all rows.
+
+Examples:
+```sql
+-- Update the item_name for all rows with a specific item_id
+UPDATE Inventory SET item_name = 'new name' WHERE item_id = 1;
+```
+
+### SET
+
+```ebnf
+SET var ( TO | '=' ) literal
+```
+
+Updates the value of a system variable.
+
+### SHOW
+
+```ebnf
+SHOW var
+```
+
+Returns the value of a system variable.
+
+## Data types
+
+SpacetimeDB data types are defined by [SATS],
+the Spacetime Algebraic Type System,
+and tuples are stored or encoded using [BSATN],
+the Binary Spacetime Algebraic Type Notation format.
+
+Spacetime SQL however does not support all of [SATS].
+In particular it has limited support for [products] and [sums].
+The language itself does not provide a way to construct them,
+nor does it provide scalar operations on top of them,
+the one exception being equality comparison.
+However the language does allow for reading them from tables,
+as well as returning them from queries.
+
+## Literals
+
+```ebnf
+literal = INTEGER | FLOAT | STRING | HEX | TRUE | FALSE ;
+```
+
+For each of the following [SATS] data types,
+we describe how to express their literal values in Spacetime SQL.
+
+### Bool
+
+In Spacetime SQL, the [boolean] Algebraic Type,
+representing the two truth values of boolean logic,
+is expressed using the canonical atoms `true` or `false`.
+
+### Integer
+
+The construction of [integer] values in Spacetime SQL.
+The concrete [SATS] type is inferred from the context.
+
+```ebnf
+INTEGER
+    = [ '+' | '-' ] NUM
+    | [ '+' | '-' ] NUM 'E' [ '+' | '-' ] NUM
+    ;
+
+NUM
+    = DIGIT { DIGIT }
+    ;
+
+DIGIT
+    = 0..9
+    ;
+```
+
```

Comment:
I think we need to add a mapping here, and elsewhere when type inference is done, meaning that in fact we support `TINYINT, INTEGER, BIGINT, ` etc


| Sats Type    | SQL Type     | Description                                   |
|--------------|--------------|-----------------------------------------------|
| `i8`         | `TINYINT`    | 8-bit signed integer                          |
| `u8`         | `UNSIGNED TINYINT` | 8-bit unsigned integer                |
| `i16`        | `SMALLINT`   | 16-bit signed integer                         |
| `u16`        | `UNSIGNED SMALLINT` | 16-bit unsigned integer              |
| `i32`        | `INT`        | 32-bit signed integer                         |
| `u32`        | `UNSIGNED INT` | 32-bit unsigned integer                  |
| `i64`        | `BIGINT`     | 64-bit signed integer                         |
| `u64`        | `UNSIGNED BIGINT` | 64-bit unsigned integer                |
| `i128`       | Not directly supported | 128-bit signed integer (handled via custom logic or string in SQL) |
| `u128`       | Not directly supported | 128-bit unsigned integer (handled via custom logic or string in SQL) |

Reply:
SATS specifies all of this, so I didn't want to duplicate. Instead I just refer to SATS. This is mainly to keep the docs concise and reduce bloat.

Reply:
Clarifying note: Mario notes this table was important because we used to interpret these SQL types in `CREATE TABLE`. Now that we no longer support DDL, it's not as important to present this mapping.